tools/OMIcheck/OMIcheck.ps1

```diff
@@ -63,15 +72,31 @@ foreach ($sub in $subs)
                     Write-Host -ForegroundColor Red `t`t $VM.Name ": VM has vulnerable OMI version " $pkgVer
                     if ($upgradeOMI)
                     {
-                        $upgrade = Invoke-AzVMRunCommand -ResourceGroupName $VM.ResourceGroupName -Name $VM.Name -CommandId 'RunShellScript' -ScriptPath $upgradeScriptPath
-                        Write-Host -ForegroundColor Red `t`t $VM.Name ": Result of OMI package upgrade attempt: " $upgrade.Value.Message
+                        # If OMSLinuxAgent is installed, nudge the Azure Guest Agent to pickup the latest bits.
+                        $VMobj = Get-AzVM -ResourceGroupName $VM.ResourceGroupName -Name $VM.Name
+                        $omsExt = $VMobj.Extensions | where { ($_.Publisher -eq "Microsoft.EnterpriseCloud.Monitoring" -and $_.VirtualMachineExtensionType -eq "OmsAgentForLinux") }
+                        if ($omsExt -ne $null)
+                        {
+                            # Trigger a goal state change for GA.
+                            Set-AzVMExtension -VMName $VM.Name -ResourceGroupName $VM.ResourceGroupName -Location $VM.Location -Publisher $omsExt.Publisher -Name $omsExt.Name -ExtensionType $omsExt.VirtualMachineExtensionType
```

Comment:
Why not set version as 1.13.40, will it necessarily upgrade if left unspecified?

Reply:
Also, should we run this `--no-wait` (or w/e the powershell equivalent is)? It's a trade-off between seeing the script output the result of each operation and the script running much faster

Reply:
I had tried setting that arg. -Version "1.13.40" and it gave me 400 bad request with ErrorMessage: The value of parameter typeHandlerVersion is invalid.

The error message didn't make sense.  Now, I tried "1.13" only and that proceeded.

Reply:
RE: -no-wait (-NoWait in PS) would be fire&forget. Customer would need to run the script second time to validate update took place.

Another concern is ARM throttling on number of operations. Without wait we can submit 40-50 operations per minute. That is withing boundary of 1200 ops/hour.

Reply:
yep good point about ARM rate limits

Reply:
> I had tried setting that arg. -Version "1.13.40" and it gave me 400 bad request with ErrorMessage: The value of parameter typeHandlerVersion is invalid.
> 
> The error message didn't make sense. Now, I tried "1.13" only and that proceeded.

Cool, IME (but this was a while ago) running without version specification would do nothing, but if you were able to test this and see otherwise, that's perfect.